core/src/test/scala/org/apache/spark/sql/delta/ShowTableColumnsSuite.scala

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta
+
+import org.apache.spark.sql.delta.test.DeltaSQLCommandTest
+import org.apache.spark.sql.functions.struct
+import org.apache.spark.sql.{AnalysisException, DataFrame, QueryTest, Row}
+import org.apache.spark.sql.test.SharedSparkSession
+import org.apache.spark.util.Utils
+
+import java.io.File
+
+trait ShowTableColumnsSuiteBase
+    extends QueryTest
+    with SharedSparkSession
+    with DeltaSQLCommandTest
+    with DeltaTestUtilsForTempViews {
+
+  import testImplicits._
+
+  private val outputColumnNames = Seq("columnName")
+  private val outputColumnValues = Seq(Seq("column1"), Seq("column2"))
+
+  protected def checkResult(
+      result: DataFrame,
+      expected: Seq[Seq[Any]],
+      columns: Seq[String]
+  ): Unit = {
+    checkAnswer(
+      result.select(columns.head, columns.tail: _*),
+      expected.map { x =>
+        Row(x: _*)
+      }
+    )
+  }
+
+  private def showDeltaColumnsTest(
+      f: File => String,
```

Comment:
nit: have a better name for `f`. ex. `fileToTableNameMapper`

Reply:
Fixed